siad/main.go

```diff
@@ -111,8 +113,10 @@ func startEnvironment(*cobra.Command, []string) {
 	}
 
 	// join the network
+	//
+	// TODO: This should proabably happen in newDaemon.
 	if !config.Siacore.NoBootstrap {
-		go d.bootstrap()
+		go d.gateway.Bootstrap(modules.BootstrapPeers[0])
```

Comment:
I guess that's fine too. I mean, when `daemon.New` initializes all the modules, their `New` functions are often doing additional internal setup anyway.